docs/spec/v1alpha1/gitrepositories.md

```diff
@@ -0,0 +1,216 @@
+# Git Repositories
+
+The `GitReposiory` API defines a source for artifacts coming from Git. 
+
+## Specification
+
+Git repository spec:
+
+```go
+// GitRepositorySpec defines the desired state of GitRepository
+type GitRepositorySpec struct {
+	// +kubebuilder:validation:Pattern="^(http|https|ssh)://"
+
+	// The repository URL, can be a HTTP or SSH address.
+	Url string `json:"url"`
+
+	// The secret name containing the Git credentials
+	// +optional
+	SecretRef *v1.LocalObjectReference `json:"secretRef,omitempty"`
+	
+	// The git reference to checkout and monitor for changes, defaults to master branch.
+	// +optional
+	Reference *GitRepositoryRef `json:"ref,omitempty"`
+
+	// The interval at which to check for repository updates.
+	Interval metav1.Duration `json:"interval"`
+}
+
+// GitRepositoryRef defines the git ref used for pull and checkout operations
+type GitRepositoryRef struct {
+	// The git branch to checkout, defaults to master.
+	// +optional
+	Branch string `json:"branch"`
+
+	// The git tag to checkout, takes precedence over branch.
+	// +optional
+	Tag string `json:"tag"`
+
+	// The git tag semver expression, takes precedence over tag.
+	// +optional
+	SemVer string `json:"semver"`
+
+	// The git commit sha to checkout, if specified branch and tag filters will be ignored.
+	// +optional
+	Commit string `json:"commit"`
+}
+```
+
+Git repository status:
+
+```go
+// GitRepositoryStatus defines the observed state of GitRepository
+type GitRepositoryStatus struct {
+	// +optional
+	Conditions []RepositoryCondition `json:"conditions,omitempty"`
+
+	// LastUpdateTime is the timestamp corresponding to the last status
+	// change of this repository.
+	// +optional
+	LastUpdateTime *metav1.Time `json:"lastUpdateTime,omitempty"`
+
+	// URI for the artifacts of the last successful repository sync.
+	// +optional
+	Artifacts string `json:"artifacts,omitempty"`
+}
+```
+
+Git repository status conditions:
+
+```go
+// RepositoryCondition contains condition information for a repository
+type RepositoryCondition struct {
```

Comment:
We need to either make this field common / shared (and probably rename it to `SourceCondition` to not assume a source type), or make it source specific (and rename it to `GitRepositoryCondition`).

Reply:
Addressed in https://github.com/fluxcd/source-controller/commit/8cd8d8f6f23798e81baf49817741647aa4315358

TLDR; I've renamed it to `SourceCondition` but this is just a temporary change until the `Condition` type makes it in Kubernetes meta API. When that happens we should drop our type and use the meta one.